core/sdk-core/src/main/java/software/amazon/awssdk/core/RequestOverrideConfiguration.java

```diff
@@ -192,19 +204,19 @@ default B rawQueryParameter(String name, String value) {
     }
 
     protected abstract static class BuilderImpl<B extends Builder> implements Builder<B> {
-        private Map<String, List<String>> headers;
+        private Map<String, List<String>> headers = new HashMap<>();
 
-        private Map<String, List<String>> rawQueryParameters;
+        private Map<String, List<String>> rawQueryParameters = new HashMap<>();
 
-        private List<ApiName> apiNames;
+        private List<ApiName> apiNames = new ArrayList<>();
 
         protected BuilderImpl() {
         }
 
         protected BuilderImpl(RequestOverrideConfiguration sdkRequestOverrideConfig) {
-            sdkRequestOverrideConfig.headers().ifPresent(this::headers);
-            sdkRequestOverrideConfig.rawQueryParameters().ifPresent(this::rawQueryParameters);
-            sdkRequestOverrideConfig.apiNames().ifPresent(apiNames -> apiNames.forEach(this::addApiName));
+            this.headers = new HashMap<>(sdkRequestOverrideConfig.headers);
```

Comment:
Are we returning mutable maps in the getters? Is this consistent with how we do models?

Reply:
No, it will create an immutable map in the `build` method. 
```java
    protected RequestOverrideConfiguration(Builder<?> builder) {
        this.headers = CollectionUtils.deepUnmodifiableMap(builder.headers(), () -> new TreeMap<>(String.CASE_INSENSITIVE_ORDER));
        this.rawQueryParameters = CollectionUtils.deepUnmodifiableMap(builder.rawQueryParameters());
        this.apiNames = Collections.unmodifiableList(builder.apiNames());
}
```

We can't use immutable map in this particular method because this method is called from `toBuilder()` and we can't add more items if it is an immutable collection.

Looking at this file https://github.com/aws/aws-sdk-java-v2/blob/master/codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/alltypesresponse.java, looks like we are not using immutable collections at all in the models... 😓

Reply:
It seems like we are doing a direct assignment to the map in the builder meaning any modifications they make to the original map would affect the builder. This may be surprising and is inconsistent with how we do it in models so I'd suggest copying the values into a new array and also returning an unmodifiable wrapper of the mapper in the builder getter.

Reply:
They can't make any modifications to the original map though because the original map is an unmodifiableMap. But yeah, sure, I'll update it to be consistent with the models.